content/develop/start/truffle.md

```diff
@@ -1,5 +1,180 @@
+
+### Mint
+
+Minter mints and transfers NFT tokens for the vaccine program participant-1. In this example, the new participant address is `0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe` and the minter address is `0x6A33382de9f73B846878a57500d055B981229ac4`.
+
+Please make sure that you are using the same deployer address as a minter address otherwise the `mint` transaction will revert.
+
+```bash
+% truffle console --network aurora
+truffle(aurora)> const cvt = await CovidVaccineToken.deployed()
+truffle(aurora)> await cvt.minter()
+'0x6A33382de9f73B846878a57500d055B981229ac4'
+truffle(aurora)> await cvt.mint('0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe')
+```
+
+You should notice that participants are not allowed to transfer their NFT tokens except for the minter. 
+
+So lets try to use any participant address to validate this. First you should exit from the 
+truffle terminal then in `truffle-config.js`, change the value of `from` field to a new address (e.g the participant address `0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe`) in the `aurora` network configuration. 
+
+```
+{
+    ...
+    aurora: {
+      provider: () => setupWallet('https://testnet.aurora.dev'),
+      network_id: 0x4e454153,
+      gas: 10000000,
+      from: '0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe'
+    },
+}
+```
+
+Then connect your Truffle console to the Aurora testnet again:
+
+```bash
+% truffle console --network aurora
+truffle(aurora)> const cvt = await CovidVaccineToken.deployed()
+truffle(aurora)> await cvt.safeTransferFrom('0x2531a4d108619a20acee88c4354a50e9ac48ecfe', '0x8722C88e82AbCC639148Ab6128Cd63333B2Ad771', 1) 
+Uncaught:
+Error: Unknown address - unable to sign transaction for this address: "0x2531a4d108619a20acee88c4354a50e9ac48ecfe"
+...
+reason: 'Invalid Transfer',
+  hijackedStack: 'Error: Unknown address - unable to sign transaction for this address: "0x2531a4d108619a20acee88c4354a50e9ac48ecfe"\n' 
+```
+This is exactly the same error message we have in our NFT contract in `safeTransferFrom`: 
+```javascript
+function safeTransferFrom(
+    address from, 
+    address to, 
+    uint256 tokenId
+) 
+    public 
+    virtual 
+    override 
+    {
+        require(
+            minter == msg.sender || to == minter,
+            'Invalid Transfer'
+        );
+        safeTransferFrom(from, to, tokenId, "");
+    }
+```
+You should change the `from` back to the minter address before continue to the next step.
+### Transfer
+
+Participant can transfer the token to the `minter` after receiving the vaccine. As shown below a participant can only send the NFT token if the receiver for this token is the minter (`0x6A33382de9f73B846878a57500d055B981229ac4`). 
+
+```bash
+% truffle console --network aurora
+truffle(aurora)> const cvt = await CovidVaccineToken.deployed()
+truffle(aurora)> await cvt.ownerOf(1)
+truffle(aurora)> await cvt.safeTransferFrom('0x2531a4D108619a20ACeE88C4354a50e9aC48ecfe', '0x6A33382de9f73B846878a57500d055B981229ac4', 1)
```

Comment:
I'm stuck here now. I get the following Error:

```
Uncaught Error: execution reverted:
    at evalmachine.<anonymous>:1:13
    at evalmachine.<anonymous>:2:49
    at sigintHandlersWrap (vm.js:272:15)
    at Script.runInContext (vm.js:127:14)
    at runScript (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle/build/webpack:/packages/core/lib/console.js:270:1)
    at Console.interpret (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle/build/webpack:/packages/core/lib/console.js:285:1)
    at bound (domain.js:427:14)
    at REPLServer.runBound [as eval] (domain.js:440:12)
    at REPLServer.onLine (repl.js:777:10)
    at REPLServer.emit (events.js:314:20)
    at REPLServer.EventEmitter.emit (domain.js:483:12)
    at REPLServer.Interface._onLine (readline.js:329:10)
    at REPLServer.Interface._line (readline.js:658:8)
    at REPLServer.Interface._ttyWrite (readline.js:1003:14)
    at REPLServer.self._ttyWrite (repl.js:867:9)
    at ReadStream.onkeypress (readline.js:205:10)
    at ReadStream.emit (events.js:314:20)
    at ReadStream.EventEmitter.emit (domain.js:483:12)
    at emitKeys (internal/readline/utils.js:335:14)
    at emitKeys.next (<anonymous>) {
  reason: 'ERC721: transfer caller is not owner nor approved',
  hijackedStack: 'Error: execution reverted:\n' +
    '    at /mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/web3-provider-engine/subproviders/provider.js:18:1\n' +
    '    at t.callback [as onreadystatechange] (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/web3/node_modules/web3-providers-http/src/index.js:96:1)\n' +
    '    at t._a [as dispatchEvent] (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:27:61)\n' +
    '    at t.dispatchEvent [as _setReadyState] (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/xhr2-cookies/dist/xml-http-request.js:208:1)\n' +
    '    at t._setReadyState [as _onHttpResponseEnd] (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/xhr2-cookies/dist/xml-http-request.js:318:1)\n' +
    '    at IncomingMessage._onHttpResponseEnd (/mnt/birchmd/tmp/aurora-examples/truffle/erc721-example/node_modules/truffle-hdwallet-provider/dist/webpack:/truffle-hdwallet-provider/Users/gnidan/src/work/truffle/node_modules/xhr2-cookies/dist/xml-http-request.js:289:47)\n' +
    '    at IncomingMessage.emit (events.js:326:22)\n' +
    '    at IncomingMessage.EventEmitter.emit (domain.js:506:15)\n' +
    '    at endReadableNT (_stream_readable.js:1241:12)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:84:21)'
}
```

I did change the `truffle-config.js` back to the minter as instructed. I'm not sure what I am doing wrong.

Reply:
I've resolved this in the examples repo. Please make pull and try it again.